__tests__/app.test.js

```diff
@@ -45,4 +45,51 @@ describe("app", () => {
         });
     });
   });
+  describe("/api/reviews", () => {
+    test("status: 200", () => {
+      return request(app).get("/api/reviews").expect(200);
+    });
+    test("responds with a reviews array", () => {
+      return request(app)
+        .get("/api/reviews")
+        .expect(200)
+        .then(({ body }) => {
+          expect(Array.isArray(body)).toBe(true);
+        });
+    });
+    test("all objects returned in array", () => {
+      return request(app)
+        .get("/api/reviews")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.length).toBe(13);
+        });
+    });
+    test("array contains objects with correct properties", () => {
+      return request(app)
+        .get("/api/reviews")
+        .expect(200)
+        .then(({ body }) => {
+          body.forEach((review) => {
```

Comment:
When you use a forEach, if you hand it an empty array, it skips the function you pass it and the test automatically passes, so passes without actually getting to the expect statements.
For this reason, you should pair the forEach test with an expect statement that checks the returned array has the expected length 🙂

Reply:
in the test before this, I have checked for the length of the returned array, is this not sufficient?

Reply:
> When you use a forEach, if you hand it an empty array, it skips the function you pass it and the test automatically passes, so passes without actually getting to the expect statements. For this reason, you should pair the forEach test with an expect statement that checks the returned array has the expected length 🙂

in the test before this, I have checked for the length of the returned array, is this not sufficient?